presto-main/src/test/java/com/facebook/presto/sql/gen/TestPageFunctionCompiler.java

```diff
@@ -197,6 +197,18 @@ public void testCommonSubExpressionInProjection()
         checkBlockEqual(cseResult.get(1), noCseResult2.get(0));
     }
 
+    @Test
+    public void testCommonSubExpressionDuplicatesInProjection()
+    {
+        PageFunctionCompiler functionCompiler = new PageFunctionCompiler(createTestMetadataManager(), 0);
+        try {
+            functionCompiler.compileProjections(SESSION.getSqlFunctionProperties(), ImmutableList.of(ADD_X_Y, ADD_X_Y), true, Optional.empty());
```

Comment:
Can we check for specific results as well (correctness)?

Reply:
Done.